File: vs2003_cd01/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/Samples/Technologies/Remoting/VB/Basic/RemotingHello/Client/ConsoleClientAll/HelloNew3.vb
﻿
Imports System
Imports System.Runtime.Remoting
Imports System.Runtime.Remoting.Channels
Imports System.Runtime.Remoting.Channels.Http
Imports Microsoft.VisualBasic

Imports Hello



Public Class HelloNew
    
    Public Shared Sub Main()
        Console.WriteLine(".NET Remoting Sample - HelloNew3")
        Console.WriteLine("--------------------------------" + ControlChars.Cr)
        Dim name As String = "Bill"
        Dim configFilename As String = "HelloNew3.exe.config"
        Dim type As Type = GetType(HelloService)
        Dim url As String = "http://localhost/RemotingHello/HelloService.soap"
        
        Console.WriteLine("Configuring Remoting from {0}", configFilename)
        RemotingConfiguration.Configure(configFilename)
        
        Console.WriteLine("Loading HttpChannel")
        ChannelServices.RegisterChannel(New HttpChannel())
        
        Console.WriteLine("Configuring HelloService as WellKnown Client Type")
        RemotingConfiguration.RegisterWellKnownClientType(type, url)
        
        Console.WriteLine("Obtaining Proxy for HelloService, using new")
        Dim helloService As New HelloService()
        
        Console.WriteLine("Checking if helloservice is really a proxy")
        If RemotingServices.IsTransparentProxy(helloService) = True Then
            Console.WriteLine("helloService is a proxy!")
        Else
            Throw New Exception("helloService is not a proxy")
        End If 
        Console.WriteLine("Calling HelloService.ReturnGreeting({0})", name)
        Dim greeting As String = helloService.ReturnGreeting(name)
        
        Console.WriteLine("HelloService.ReturnGreeting returned: {0}", greeting)
    End Sub 'Main
End Class 'HelloNew

File: vs2003_cd01/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/Samples/Technologies/Remoting/Basic/RemotingHello/Client/consoleclientall/HelloNew3.cs
/*=====================================================================

  File:      HelloNew3.cs

---------------------------------------------------------------------
This file is part of the Microsoft .NET Framework SDK Code Samples.

  Copyright (C) Microsoft Corporation.  All rights reserved.

This source code is intended only as a supplement to Microsoft
Development Tools and/or on-line documentation.  See these other
materials for detailed information regarding Microsoft code samples.
 
THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY
KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
PARTICULAR PURPOSE.
=====================================================================*/

using System;
using System.Runtime.Remoting;
using System.Runtime.Remoting.Channels;
using System.Runtime.Remoting.Channels.Http;

using Hello;

public class HelloNew
{
    public static void Main(String[] args)
    {
        Console.WriteLine(".NET Remoting Sample - HelloNew3");
        Console.WriteLine("--------------------------------\n");
        String name = "Bill";    
        String configFilename = "HelloNew3.exe.config";
        Type type = typeof(HelloService);
        String url = "http://localhost/RemotingHello/HelloService.soap";

        Console.WriteLine("Configuring Remoting from {0}", configFilename);
        RemotingConfiguration.Configure(configFilename);

        Console.WriteLine("Loading HttpChannel");
        ChannelServices.RegisterChannel(new HttpChannel());

        Console.WriteLine("Configuring HelloService as WellKnown Client Type");
        RemotingConfiguration.RegisterWellKnownClientType(type, url); 

        Console.WriteLine("Obtaining Proxy for HelloService, using new");
        HelloService helloService = new HelloService();

        Console.WriteLine("Checking if helloservice is really a proxy");
        if (RemotingServices.IsTransparentProxy(helloService) == true)
            Console.WriteLine("helloService is a proxy!");
        else
            throw   new Exception("helloService is not a proxy");

        Console.WriteLine("Calling HelloService.ReturnGreeting({0})", name);
        String greeting = helloService.ReturnGreeting(name);

        Console.WriteLine("HelloService.ReturnGreeting returned: {0}", greeting);
    }
}
